server/src/connectionHandlers/onReferences.ts

```diff
@@ -0,0 +1,60 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) 2023 Savoir-faire Linux. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import type * as LSP from 'vscode-languageserver/node'
+import { logger } from '../lib/src/utils/OutputLogger'
+import { analyzer } from '../tree-sitter/analyzer'
+
+export function onReferencesHandler (referenceParams: LSP.ReferenceParams): LSP.Location[] | null {
+  const { textDocument: { uri }, position } = referenceParams
+
+  const wordPosition = {
+    line: position.line,
+    character: Math.max(position.character - 1, 0)
+  }
+
+  const word = analyzer.wordAtPointFromTextPosition({
+    ...referenceParams,
+    position: wordPosition
+  })
+
+  if (word === null) {
+    return null
+  }
+
+  logger.debug(`[onReferences] Position: Line ${position.line} Character ${position.character}`)
+  logger.debug(`[onReferences] Word: ${word}`)
+
+  // TODO: this following part is the same the one in onDefinition, we should refactor it
+  const references: LSP.Location[] = []
+  const canProvideGoToDefinitionForSymbol = analyzer.isIdentifierOfVariableAssignment(referenceParams) ||
+      (analyzer.isVariableExpansion(uri, position.line, position.character) && analyzer.isIdentifier(referenceParams))
+      // Variables in declartion and variable expansion syntax
+  if (canProvideGoToDefinitionForSymbol) {
+    analyzer.getExtraSymbolsForUri(uri).forEach((globalDeclaration) => {
+      if (globalDeclaration[word] !== undefined) {
```

Comment:
This might be more straightforward with `globalDeclaration[word]?.forEach` (same with ownSymbol)

Reply:
closed for it is outdated